src/Profiling.cpp

```diff
@@ -88,12 +223,29 @@ Stmt inject_profiling(Stmt s, string pipeline_name) {
 
     Expr get_state = Call::make(Handle(), "halide_profiler_get_state", {}, Call::Extern);
 
+    Expr get_pipeline_state = Call::make(Handle(), "halide_profiler_get_pipeline_state", {pipeline_name}, Call::Extern);
+
     Expr profiler_token = Variable::make(Int(32), "profiler_token");
 
     Expr stop_profiler = Call::make(Int(32), Call::register_destructor,
                                     {Expr("halide_profiler_pipeline_end"), get_state}, Call::Intrinsic);
 
+    int stack_peak = 0;
+    for (const auto& iter : profiling.func_stack_peak) {
+        stack_peak = std::max(stack_peak, iter.second);
```

Comment:
After discussion in person, we realized we really can't predict peak stack usage of the pipeline as a whole - we can just report each Func that goes on the stack and how much space it uses. So cut this